src/seccomp/src/bin.rs

```diff
@@ -0,0 +1,567 @@
+                        "args": [
+                            {
+                                "arg_index": 3,
+                                "arg_type": "DWORD",
+                                "op": "Eq",
+                                "val": 65
+                            }
+                        ]
+                    }
+                ]
+            }
+        }
+        "#
+        .to_string()
+    }
+
+    #[test]
+    fn test_error_messages() {
+        let path = PathBuf::from("/path");
+        assert_eq!(
+            format!(
+                "{}",
+                Error::Bincode(BincodeError::new(bincode::ErrorKind::SizeLimit))
+            ),
+            format!(
+                "Bincode (de)serialization failed: {}",
+                BincodeError::new(bincode::ErrorKind::SizeLimit)
+            )
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                Error::FileFormat(FilterFormatError::MultipleSyscallFields)
+            ),
+            format!("{}", FilterFormatError::MultipleSyscallFields)
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                Error::FileOpen(path.clone(), io::Error::from_raw_os_error(2))
+            ),
+            format!(
+                "Failed to open file {:?}: {}",
+                path,
+                io::Error::from_raw_os_error(2)
+            )
+            .replace("\"", "")
+        );
+        assert_eq!(format!("{}", Error::InvalidArch), "Invalid arch");
+        assert_eq!(
+            format!(
+                "{}",
+                Error::JSON(serde_json::from_str::<serde_json::Value>("").unwrap_err())
+            ),
+            format!(
+                "Error parsing JSON: {}",
+                serde_json::from_str::<serde_json::Value>("").unwrap_err()
+            )
+        );
+        assert_eq!(format!("{}", Error::MissingInputFile), "Missing input file");
+    }
+    #[test]
+    fn test_get_argument_values() {
+        let arg_parser = build_arg_parser();
+        let default_out_file_name = "bpf.out";
+        // correct arguments
+        let arguments = &mut arg_parser.arguments().clone();
+        arguments
+            .parse(
+                vec![
+                    "seccompiler",
+                    "--input_file",
+                    "foo.txt",
+                    "--target_arch",
+                    "x86_64",
+                ]
+                .into_iter()
+                .map(String::from)
+                .collect::<Vec<String>>()
+                .as_ref(),
+            )
+            .unwrap();
+        assert_eq!(
+            get_argument_values(arguments).unwrap(),
+            Arguments {
+                input_file: "foo.txt".to_string(),
+                output_file: default_out_file_name.to_string(),
+                target_arch: "x86_64".to_string()
+            }
+        );
+
+        let arguments = &mut arg_parser.arguments().clone();
```

Comment:
I would also add cases when you do not provide values to `--input_file`, `--target-arch` and `--output_file` as well. WDYT?

Reply:
> I would also add cases when you do not provide values to --input_file, --target-arch and --output_file as well. WDYT?

I didn't provide test cases for these since they are either `required` or have default argument values. However, I will add all the above test cases, and will try to think of some more. Would be nice to have the extra assurance

Reply:
added test cases for: missing `--input-file`, missing `--target-arch` and inexistent input file